lib/portlayer/exec/exec.go

```diff
@@ -224,6 +223,11 @@ func WaitForContainerStop(ctx context.Context, id string) error {
 	Config.EventManager.Subscribe(topic, sub, watch)
 	defer Config.EventManager.Unsubscribe(topic, sub)
 
+	// Container has changed it's status, existing.
+	if c.CurrentState() == StateStopped {
```

Comment:
I don't think we can miss anything here. We are already subscribed to events, thus it is only needed to avoid the case you submitted bug about.